src/corehost/corehost.cpp

```diff
@@ -33,9 +34,18 @@
 #define EMBED_HASH_HI_PART_UTF8 "c3ab8ff13720e8ad9047dd39466b3c89" // SHA-256 of "foobar" in UTF-8
 #define EMBED_HASH_LO_PART_UTF8 "74e592c2fa383d4a3960714caef0c4f2"
 #define EMBED_HASH_FULL_UTF8    (EMBED_HASH_HI_PART_UTF8 EMBED_HASH_LO_PART_UTF8) // NUL terminated
+
+bool is_exe_enabled_for_execution_hack(pal::string_t* app_dll)
+{
+	pal::string_t host_path;
+	pal::get_own_executable_path(&host_path);
+	*app_dll = strip_executable_ext(get_filename(host_path)).append(_X(".dll"));
```

Comment:
We should consider if it's worth to support renaming the bundle. As is, it cannot ever be renamed, otherwise it won't run. That is a relatively unexpected property for an executable.